GTAP parsing into dev

CamiCit opened the pull request:
It is possible to parse GTAP in GDX and CSV (less complete) format. Includes VY matrix. V and E matrixes are originally multidimensional, so V and E rows contain regional and sectorial data.

CamiCit committed:
GTAP-MRIO parser first implementation

LorenzoRinaldi committed:
fixed paths

CamiCit committed:
Changes in function missing

CamiCit committed:
Merge branch 'dev_GTAP' of https://github.com/it-is-me-mario/MARIO into dev_GTAP

LorenzoRinaldi committed:
fixed indices in matrices at the end of parsing

LorenzoRinaldi committed:
added some prints

CamiCit committed:
Change IMTAX code

Changed row_name_categ=ITX (was MTX, conflicting with other account MTAX)

CamiCit committed:
GTAP gdx parser

Added new GTAP parser from gdx files. Old parse_GTAP is now parse_GTAP_csv

CamiCit committed:
Fix in GTAP gdx and csv parsers

CSV parser: error in ptax matrix parsing
CSV and GDX: change of denomination from "dom" to "dms" to avoid confusion with Nation names.

CamiCit committed:
GTAP CSV parser: names fix

In the CSV, the factors of production names were of 3 letters instead of 4. Now coherent with GDX parser.

Another minor variables name fix between V and Z (no consequence on output)

CamiCit committed:
Fix parser GDX: EY missing rows

When building EY, the rows for Process Emissions (even though empty) must be included.

CamiCit committed:
Adapting to new GTAP version of GSDFEMI.GDX

Change in GTAP gdx parser for emissions

CamiCit committed:
Added VY parsing

Only added the final step, VY dataframe was already built in previous work.

mohammadamint commented on mario/tools/tableparser.py:
```diff
@@ -1706,3 +1709,1059 @@ def parser_figaro_e3(path,doping_value):
+            return matrix
+
+    # Matrix Z and Y
+    print('Starting Z and Y')
```
please avoid direct printing. In case you need to print something it needs to be done visa log_time function

mohammadamint commented on mario/tools/tableparser.py:
```diff
@@ -1706,3 +1709,1059 @@ def parser_figaro_e3(path,doping_value):
+            _ENUM['E']: E,
+            _ENUM['EY']: EY,
+            _ENUM['X']: X,  
+        }
```
you can pas VY matrix also here, and the rest will be done by itself

mohammadamint commented on mario/tools/tableparser.py:
```diff
@@ -1706,3 +1709,1059 @@ def parser_figaro_e3(path,doping_value):
+    tuple
+        (matrices, indeces, units)
+    """
+    from gams import transfer as gt
```
bring the imports to the beginning of code

mohammadamint commented on mario/tools/tableparser.py:
```diff
@@ -1706,3 +1709,1059 @@ def parser_figaro_e3(path,doping_value):
+
+    #Satellite account
+    # Filter the data for 'Emissions': Emi=from combustion Emi_proc=from processes
+    print("Stating: E and EY")
```
use log function

mohammadamint commented on test_GTAP.py:
```diff
@@ -0,0 +1,22 @@
+#%%
+import mario
```
this file needs to be removed completely from the github repo

mohammadamint committed:
Merge branch 'dev' into dev_GTAP

CamiCit committed:
Making GTAP parsing faster

1)Missing function is improved to make it faster
2)Some functions removed from parser_gtap_mario_gdx into GTAP_utils.py
3)Added logs
4)Added exceptions

CamiCit committed:
Change add_sector routine to prepare disaggregation changes

Distinction between add or split in master file. Added new sheets for Output, Exclusions and Trades. Added some sanity check when getting and reading the inventories.

CamiCit committed:
Start of new_sectors.py in tools (and more sanity check)

Sanity check moved to excelhandler.py instead of AttrData.py and expanded.
new_sectors.py file created inside tools: will be the place where disaggregation takes place, after add_sectors has already created new coefficient columns. First function: _new_flow_columns (obtains new X and computes Z,V,E)

CamiCit committed:
Flows computation and Uncertainty matrix

In new_sectors.py computation of flow matrices from the updated coefficients, when splitting.
Implementation of uncertainty matrix: pre-defined constants, possibility to define new coefficients (uncertainty_values), and definition of uncertainty_matrix, automatically filled according to given data characteristics (clusters or not).

LorenzoRinaldi committed:
removed test_gtap script.py

LorenzoRinaldi committed:
Bug fixed: developments for adding sectors to IOTs causing issues for SUTs

CamiCit committed:
Splitting sectors: new flows rows and cols + Flat export including 0

Created _row_hypothesis function and integrated it in AttrData. Improvement of _new_flow_columns

In excelhandler database_txt_flat() included option for not excluding zeroes.

CamiCit committed:
First integration of cvxlab (for sector splitting)

Added cvxlab integration to the sector splitting workflow.
New folder in tools where default cases can be inserted as new folders (now present Split_sectors and SRIO) including model_settings and a mapping from mario to cvxlab sets. Included in AttrData.Database.add_sectors new parameters, sanity checks, and invocation of cvxlab optimization routines.
Also fixed a bug in new_sectors.py to deepcopy matrices when creating split scenarios.
Also updated constants for trade and exclusion sheet columns.

CamiCit committed:
Integration of cvxlab output into add_sector

First draft of integration of cvxlab output database into MARIO tables. Specific function for 'Split_sectors' case, still generic for other cases.
To be polished.

CamiCit committed:
Minor fix for cvxlab optimization

Exporting flat matrices including zeroes

CamiCit committed:
Tolerance definition from inventory

Tolerance for disaggregation routines can be defined by the user in the add_sectors inventory, a new sheet was created. There is a problem with the export of the split matrices, will be fixed in next push.
